### ISSUE TITLE
prometheus query time alignment cause bar width to fluctuate

### ISSUE THREAD

**Issue by roidelapluie** [grafana/grafana#12379]
![bug](https://user-images.githubusercontent.com/291750/41768162-c127fe70-760b-11e8-9219-d9b0f0bb07d6.png)

to reproduce: create a bar graph panel

zoom it zoom out move forward ... after a few attemps the problem shows.

in real condition it shows up randomly.

**Comment by dswarbrick** [comment]
I just noticed this too, but strangely it only affects data from one particular host so far. I am using node_exporter 0.16.0, and am graphing CPU times by state (e.g. user, system, iowait etc). I only notice this weird effect for a host that has 64 cores, but I really doubt it's related to that.

**Comment by torkelo** [comment]
Can you show the query used to replicate this?

**Comment by dswarbrick** [comment]
I have two drop-downs / variables, executing the `label_values()` function to populate `$instance` and `$cpu`. The panel query is as follows:
```
rate(node_cpu_seconds_total{instance="$instance", cpu="$cpu", mode!="idle"}[1m])
```

**Comment by davkal** [comment]
I cant repro this on master. Tried Firefox and Chrome. Digging a bit through the code...

**Comment by davkal** [comment]
Found a good repro:

- robust perception data source
- bar chart
- query: `prometheus_engine_query_duration_seconds > 0.0001`
- set times from `2018-07-02 13:01:30` to `2018-07-02 13:06:30`

-> all normal

- set times from `2018-07-02 13:01:29` to `2018-07-02 13:06:29`

-> slim bars

**Comment by dswarbrick** [comment]
@davkal I think it depends on what time you initially open the dashboard, which is why it seems a bit hit and miss. If you're lucky enough to hit a multiple of $__interval, everything is fine. As I mentioned earlier however, simply clicking in the manual dashboard refresh several times lets you see the whole spectrum of bar widths ;-)

**Comment by davkal** [comment]
Yup, but the whole point of alignment is that it should not matter when you open the dashboard. I'm debugging this now...

**Comment by FUSAKLA** [comment]
Hi, is there any planned date of releasing this fix? I see that 5.2.2 milestone is complete.
Charts using bars are kind of useless right now, so just to know if I should rollback or wait.
Thanks for any info.

**Comment by dtennander** [comment]
Hi there!
We have started to notice this issue with version `6.6.2 (3fa63cfc34)` again... I see that a fix was added in #12483 but the problem seem to have reemerged. Do you want us to open a new issue or should we reopen this? 

Our query:
```promql
round(sum by (label_app) (increase(logback_appender_total{environment="$env", level="error"}[$__interval]) * on (pod) group_left(label_app) kube_pod_labels{label_brand="asd"}))
```

Thanks for the help!

**Comment by kustodian** [comment]
@dtennander I guess it's better to open a new issue, because closed issues are rarely looked after.
